shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley/API/Validation.hs

```diff
@@ -23,19 +24,34 @@ import           Control.Monad.Except
 import           Control.Monad.Trans.Reader (runReader)
 import           Control.State.Transition.Extended (TRC (..), applySTS)
 import           Data.Either (fromRight)
+import           Data.Maybe (isNothing)
 import           GHC.Generics (Generic)
 import           Ledger.Core (Relation (..))
-import           Shelley.Spec.Ledger.BaseTypes (Globals)
+import           Shelley.Spec.Ledger.BaseTypes (Globals (..))
 import           Shelley.Spec.Ledger.BlockChain
 import qualified Shelley.Spec.Ledger.LedgerState as LedgerState
+import           Shelley.Spec.Ledger.PParams (PParams)
 import           Shelley.Spec.Ledger.Slot (SlotNo)
 import qualified Shelley.Spec.Ledger.STS.Bbody as STS
+import qualified Shelley.Spec.Ledger.STS.Chain as STS
 import qualified Shelley.Spec.Ledger.STS.Tick as STS
 import qualified Shelley.Spec.Ledger.TxData as Tx
 
 -- | Type alias for the state updated by TICK and BBODY rules
 type ShelleyState = LedgerState.NewEpochState
 
+{-------------------------------------------------------------------------------
+  CHAIN Transition checks
+-------------------------------------------------------------------------------}
+chainChecks ::
+  forall crypto.
+  (Crypto crypto) =>
+  Globals ->
+  PParams ->
+  BHeader crypto ->
+  Bool
+chainChecks globals pp bh = isNothing $ STS.chainChecks (maxMajorPV globals) pp bh
```

Comment:
Just return the error here too, we'd like to trace the reason why a header/block is invalid.

(Use `Except` here too?)

Reply:
gotcha, makes sense!